commit 04157f13de8d9b3028bca3a1832276ba96ae45fd
Author: Martin Gräßlin <mgraesslin@kde.org>
Date:   Fri Aug 22 07:37:42 2014 +0200

    [libkwineffects] Require strict binding for intel driver
    
    According to driver devs it's a "buggy micro-optimisation", see
    https://bugs.freedesktop.org/show_bug.cgi?id=80349#c1
    
    CCBUG: 336589
    
    Backport from kwin.git e8241313fc979aefafc8c2a0207570d6394f1580

diff --git a/kwin/libkwineffects/kwinglplatform.cpp b/kwin/libkwineffects/kwinglplatform.cpp
index d9eb97d..c8e5ba1 100644
--- a/kwin/libkwineffects/kwinglplatform.cpp
+++ b/kwin/libkwineffects/kwinglplatform.cpp
@@ -818,7 +818,8 @@ void GLPlatform::detect(OpenGLPlatformInterface platformInterface)
             m_supportsGLSL = false;
 
         m_limitedGLSL = m_supportsGLSL && m_chipClass < I965;
-        m_looseBinding = true;
+        // see https://bugs.freedesktop.org/show_bug.cgi?id=80349#c1
+        m_looseBinding = false;
 
         if (m_chipClass < I965) {
             m_recommendedCompositor = OpenGL1Compositing;
